### PR TITLE
Move data creation for `estimate_truncation()` to `data_raw`

### LINKED ISSUE #507: Example data set for `estimate_truncation`
At the moment an example data set is created in the [estimated_truncation examples](https://github.com/epiforecasts/EpiNow2/blob/a3ff51da886b4c243b2b3d4e8f6011639968fd8f/R/estimate_truncation.R#L96).

It would probably make sense to move this code out of the example and into https://github.com/epiforecasts/EpiNow2/tree/main/data-raw where it can be used to generate a dataset that can be shipped with the package, in https://github.com/epiforecasts/EpiNow2/tree/main/inst/extdata as is already the case for `estimate_infections` and `regional_epinow`

### PR DESCRIPTION
**Pull request by jamesmbaazam**
## Description

This PR closes #507.

This PR simplifies the example in `estimate_truncation()` by moving the data creation function elsewhere where it is made an internal function and used to create the dataset to be shipped as part of the package. The dataset is also used in the `estimate_truncation()` example. 

## Initial submission checklist

- [x] My PR is based on a package issue and I have explicitly linked it.
- [x] I have tested my changes locally (using `devtools::test()` and `devtools::check()`).
- [x] I have added or updated unit tests where necessary.
- [x] I have updated the documentation if required and rebuilt docs if yes (using `devtools::document()`).
- [x] I have followed the established coding standards (and checked using `lintr::lint_package()`).
- [x] I have added a news item linked to this PR.

## After the initial Pull Request 

- [x] I have reviewed Checks for this PR and addressed any issues as far as I am able.

### CODE REVIEW

File: R/estimate_truncation.R [hunk 1]
```diff
@@ -83,46 +83,8 @@
 #' old_opts <- options()
 #' options(mc.cores = ifelse(interactive(), 4, 1))
 #'
-#' # get example case counts
```
Comment (seabbs) [inline R/estimate_truncation.R:86]: Part of the reason to do it like this was to make it crystal clear what the assumed data generating process was. Do you think this just didn't work or if it did work we can instead improve the documentation (or perhaps we already have) to cover this gap?

File: data-raw/estimate-truncation.R [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  sd = convert_to_logsd(3, 2),
+  sd_sd = 0.1,
+  max = 10
+)
+
+# Use the make_truncated_data function to generate example data for
+# an example using estimate_truncation()
+example_truncation_data <- purrr::map(
+  c(20, 15, 10, 0),
+  make_truncated_data,
```
Comment (seabbs) [inline data-raw/estimate-truncation.R:20]: do we want this an internal function or do we want to define it here?

**Review by seabbs — COMMENTED** [review]
Nice idea. A few comments from me.

File: data-raw/estimate-truncation.R (hunk 2 above)
Comment (jamesmbaazam) [reply data-raw/estimate-truncation.R:20]: I was debating where was best to store the function and agree it should live here as the whole point of this script is to make the data generation process standalone.

**Comment by jamesmbaazam** [comment]
A slight plea to not comment on draft PRs until marked as ready for review 😬.

File: R/estimate_truncation.R (hunk 1 above)
Comment (jamesmbaazam) [reply R/estimate_truncation.R:86]: > Do you think this just didn't work or if it did work we can instead improve the documentation

I agree with the original issue that it was best placed elsewhere and linked in the example. I think it's good to keep the example focused without the user having to understand the data generation step. We can link to that as Seb has pointed out in the `./R/data.R` script.

File: NEWS.md [hunk 3]
```diff
@@ -26,6 +26,7 @@
...
 * Added the possibility of specifying a fixed observation scaling. By @sbfnk in #550 and reviewed by @seabbs.
 * Added the possibility of specifying fixed overdispersion. By @sbfnk in #560 and reviewed by @seabbs.
+* The example in `estimate_truncation()` has been simplified. The package now ships with a dataset `example_truncation_data`, which is used in the `estimate_truncation()` example. The steps for creating the truncated data is now stored in `./data-raw/estimate-truncation.R`. By @jamesmbaazam in #584 and reviewed by @<REVIEWER>.
```
Comment (jamesmbaazam) [inline NEWS.md:29]:
```suggestion
* The example in `estimate_truncation()` has been simplified. The package now ships with a dataset `example_truncation_data`, which is used in the `estimate_truncation()` example. The steps for creating the truncated data is now stored in `./data-raw/estimate-truncation.R`. By @jamesmbaazam in #584 and reviewed by @seabbs and @sbfnk.
```

File: R/estimate_truncation.R (hunk 1 above)
Comment (seabbs) [reply R/estimate_truncation.R:86]: Awesome. So action here is to add a link to the new data-raw script

**Review by sbfnk — CHANGES_REQUESTED** [review]
Nice!

- we have some `lintr` failures because we don't have `here` or `usethis` in Suggests. Perhaps we should remove `here` from the examples (and just use `file.path` assuming it's run from the package root) but add `usethis` to Suggests?
- we also have failed synthetic validation which we also see in #504 and which is unrelated
- can we also update the tests in `tests/testthat/test-estimate_truncation.R` to use the new example instead of duplicating the code to create it?

**Comment by jamesmbaazam** [comment]
> * Perhaps we should remove `here` from the examples (and just use `file.path` assuming it's run from the package root) but add `usethis` to Suggests?

They're actually [in suggests](https://github.com/epiforecasts/EpiNow2/blob/db954b80cd224256c63a8277b2f01e34002ef581/DESCRIPTION#L129) so I'm confused by the linter failure.

**Comment by sbfnk** [comment]
> They're actually [in suggests](https://github.com/epiforecasts/EpiNow2/blob/db954b80cd224256c63a8277b2f01e34002ef581/DESCRIPTION#L129) so I'm confused by the linter failure.

Ah sorry so this is due to the changes I made to accommodate the `cmdstanr` suggest which might not be available (following the example from `brms`. So it can be simply solved by adding `here` and `usethis` to `extra-packages` in `setup-r-dependencies` in the lint action.

**Comment by seabbs** [comment]
(review changes were me trying to unassign myself - i think you both have got this one in the bag)

**Comment by jamesmbaazam** [comment]
> So we need to be explicit in what we want installed and the issue here can be simply addressed by adding `here` and `usethis` to `extra-packages` in `setup-r-dependencies` in the lint action.

Nice. Should we address this in a separate issue?

**Comment by sbfnk** [comment]
> > So we need to be explicit in what we want installed and the issue here can be simply addressed by adding `here` and `usethis` to `extra-packages` in `setup-r-dependencies` in the lint action.
> 
> Nice. Should we address this in a separate issue?

Either that or otherwise also fine to add here as it's a minor change and addressing triggered by this PR (if not caused by it).

**Comment by jamesmbaazam** [comment]
> > > So we need to be explicit in what we want installed and the issue here can be simply addressed by adding `here` and `usethis` to `extra-packages` in `setup-r-dependencies` in the lint action.
> > 
> > 
> > Nice. Should we address this in a separate issue?
> 
> Either that or otherwise also fine to add here as it's a minor change and addressing triggered by this PR (if not caused by it).

I've fixed that here [0328827](https://github.com/epiforecasts/EpiNow2/pull/584/commits/03288271b2936afe6202fd9c07ac16b29dda50aa)

File: R/data.R [hunk 4]
```diff
@@ -45,3 +45,19 @@
...
+#' `estimate_truncation()`. For details on how the data is generated, see
+#' <https://github.com/epiforecasts/EpiNow2/blob/main/data-raw/estimate-truncation.R> #nolint
+#' @format A list of `data.table`s containing cases reported on each date until
+#' a point of truncation.
+#' #' Each element of the list is a `data.table` with the following columns:
+#' \describe{
+#'   \item{date}{Date of case report.}
+#'   \item{confirm}{Number of confirmed cases.}
+#' }
+"example_truncation_data"
```
[the remark this answers is not in the document]
Comment (jamesmbaazam) [reply R/data.R]: I've shortened the name in [61a5561](https://github.com/epiforecasts/EpiNow2/pull/584/commits/61a55610c7d6d05c1aec58c4e9b9ac7be3c802f9)

**Comment by jamesmbaazam** [comment]
> * can we also update the tests in `tests/testthat/test-estimate_truncation.R` to use the new example instead of duplicating the code to create it?

Thanks for spotting. I've fixed it here [d899f78](https://github.com/epiforecasts/EpiNow2/pull/584/commits/d899f780336f7e3bd2e1afa58923683bb2fcf25d).

File: tests/testthat/test-estimate_truncation.R [hunk 5]
```diff
@@ -6,39 +6,8 @@ futile.logger::flog.threshold("FATAL")
...
-  trunc_cases[(.N - length(cmf) + 1):.N, confirm := as.integer(confirm * cmf)]
-  return(trunc_cases)
-}
-example_data <- purrr::map(c(20, 15, 10, 0),
-  construct_truncation,
-  cases = reported_cases,
-  dist = trunc_dist
-)
+# get example truncated case counts
+example_data <- example_truncated
```
Comment (sbfnk) [inline tests/testthat/test-estimate_truncation.R:10]: This is not needed. Just apply the tests directly to `example_truncated` below.

File: R/estimate_truncation.R [hunk 6]
```diff
@@ -83,46 +83,9 @@
...
-#'   construct_truncation,
-#'   cases = reported_cases,
-#'   dist = trunc
-#' )
+#' # get example data stored in the package
+#' # Run `?example_truncation_data` for more details
+#' example_data <- example_truncated
 #'
 #' # fit model to example data
 #' est <- estimate_truncation(example_data,
```
Comment (sbfnk) [inline R/estimate_truncation.R:91]:
```suggestion
#' # fit model to example data
#' # See [example_truncated] for more details
#' est <- estimate_truncation(example_truncated,
```

**Review by sbfnk — CHANGES_REQUESTED** [review]

File: NEWS.md [hunk 7]
```diff
@@ -26,6 +26,7 @@
...
 * Added the possibility of specifying a fixed observation scaling. By @sbfnk in #550 and reviewed by @seabbs.
 * Added the possibility of specifying fixed overdispersion. By @sbfnk in #560 and reviewed by @seabbs.
+* The example in `estimate_truncation()` has been simplified. The package now ships with a dataset `example_truncation_data`, which is used in the `estimate_truncation()` example. The steps for creating the truncated data is now stored in `./data-raw/estimate-truncation.R`. By @jamesmbaazam in #584 and reviewed by @seabbs and @sbfnk.
```
Comment (jamesmbaazam) [inline NEWS.md:29]:
```suggestion
* The example in `estimate_truncation()` has been simplified. The package now ships with a dataset `example_truncated`, which is used in the `estimate_truncation()` example and tests. The steps for creating the `example_truncated` is stored in `./data-raw/estimate-truncation.R`. By @jamesmbaazam in #584 and reviewed by @seabbs and @sbfnk.
```

File: tests/testthat/test-estimate_truncation.R (hunk 5 above)
Comment (sbfnk) [reply tests/testthat/test-estimate_truncation.R:10]: Isn't this one still open?

Comment (jamesmbaazam) [reply tests/testthat/test-estimate_truncation.R:10]: Strange! I fixed it here 793ae679ef61576de9f28410d0535cf432c25115.

Comment (jamesmbaazam) [reply tests/testthat/test-estimate_truncation.R:10]: Sorry, I was looking at a different file.

**Review by sbfnk — APPROVED** [review]

**Pull request merged**